File: 118.pascals-triangle.go
// @leet start
func generate(numRows int) [][]int {
  res := make([][]int, numRows)
  for i := 0; i < numRows; i++ {
    for j := 0; j <= i; j++ {
      if j == 0 || j == i {
        res[i] = append(res[i], 1)
      } else if i > 1 {
        res[i] = append(res[i], res[i-1][j-1]+res[i-1][j])
      }
    }
  }
  return res
}
// @leet end

File: 118.pascals-triangle.js
// @leet start
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  let pascal = [[1]]
  for (let i = 1; i < numRows; i++) {
    let row = []
    for (let j = 0; j <= i; j++) {
      if (j === 0 || j === i) {
        row.push(1)
      }
      else if (i > 1) {
        row.push(pascal[i - 1][j - 1] + pascal[i - 1][j])
      }
    }
    pascal.push(row)
  }
  return pascal
};
// @leet end
